src/VideoDecoderWorker.cpp

```diff
@@ -80,191 +80,117 @@ void VideoDecoderWorker::RemoveVideoOutput(VideoOutput* output)
 
 int VideoDecoderWorker::Decode()
 {
-	QWORD		frameTimestamp = (QWORD)-1;
-	QWORD		frameTime = (QWORD)-1;
-	DWORD		frameClockRate = (DWORD)-1;
-	QWORD		frameSenderTime = 0;
-	DWORD		lastSeq = RTPPacket::MaxExtSeqNum;
 	bool		waitIntra = false;
 
 	Log(">VideoDecoderWorker::Decode()\n");
 
 	//Mientras tengamos que capturar
 	while(decoding)
 	{
-		//Obtenemos el paquete
-		if (!packets.Wait(0))
+		//Wait for an incoming frame
+		if (!frames.Wait(0))
 			//Done
 			break;
 
-		//Get packet in queue
-		auto packet = packets.Pop();
+		//Get frame from queue
+		auto videoFrame = frames.Pop();
 
 		//Check
-		if (!packet)
+		if (!videoFrame)
 			//Check condition again
 			continue;
 		
-		//Get extended sequence number and timestamp
-		DWORD seq = packet->GetExtSeqNum();
-		QWORD ts = packet->GetExtTimestamp();
-		QWORD time = packet->GetTime();
-		DWORD clockRate = packet->GetClockRate();
-		QWORD senderTime = packet->HasAbsoluteCaptureTime() ? packet->GetAbsoluteCaptureTime() : 0;
-
 		//If we don't have codec
-		if (!videoDecoder || (packet->GetCodec()!=videoDecoder->type))
+		if (!videoDecoder || (videoFrame->GetCodec()!=videoDecoder->type))
 		{
 			//Create new codec from pacekt
-			videoDecoder.reset(VideoCodecFactory::CreateDecoder((VideoCodec::Type)packet->GetCodec()));
+			videoDecoder.reset(VideoCodecFactory::CreateDecoder((VideoCodec::Type)videoFrame->GetCodec()));
```

Comment:
cast no longer needed, better remove it

```suggestion
			videoDecoder.reset(VideoCodecFactory::CreateDecoder(videoFrame->GetCodec()));
```